src/libsystemd/sd-journal/journal-file.c

```diff
@@ -3071,6 +3081,16 @@ static int generic_array_bisect(
 
                         for (;;) {
                                 if (left == right) {
+                                        if (m != m_original && direction == DIRECTION_DOWN) {
+                                                /* We found corrupted entries, and the final index may not be tested yet. */
```

Comment:
"We found one or more entries in `generic_array_bisect_step()`, which means `right` has been modified but we haven't found a matching object yet so if `left == right`, we still have to check the final object to see if it actually matches" ?